packages/api/src/command/webhook/webhook-request.ts

```diff
@@ -42,22 +44,41 @@ export function buildWebhookRequestData(create: CreateWebhookRequestCommand): We
   };
 }
 
-export type UpdateWebhookLogStatusCommand = {
+export type UpdateWebhookRequestCommand = {
   id: string;
   status: WebhookRequestStatus;
+  statusDetail?: string;
+  requestUrl?: string;
+  httpStatus?: number;
+  durationMillis?: number;
 };
 
-export const updateWebhookRequestStatus = async ({
+export async function updateWebhookRequest({
   id,
   status,
-}: UpdateWebhookLogStatusCommand): Promise<void> => {
-  const log = await WebhookRequest.findOne({ where: { id } });
-  if (!log) throw new NotFoundError(`Could not find webhook requst ${id}`);
-  await WebhookRequest.update({ status }, { where: { id } });
-};
+  statusDetail,
+  requestUrl,
+  httpStatus,
+  durationMillis,
+}: UpdateWebhookRequestCommand): Promise<void> {
+  const whRequest = await WebhookRequest.findOne({ where: { id } });
+  if (!whRequest) throw new NotFoundError(`Could not find webhook request ${id}`);
+  const statusDetailParsed = limitStringLength(statusDetail, MAX_VARCHAR_LENGTH);
+  const requestUrlParsed = limitStringLength(requestUrl, MAX_VARCHAR_LENGTH);
```

Comment:
is the behavior for the requestUrl ideal here? If its longer than 255 (which isnt that unlikely) we truncate it? Cant that lead to us sending webhooks to the wrong url potentially or other indeterminate behavior? If the 255 char limit is an important one, wouldnt it be better to place it in a zod schema that prevents customers from setting a WH url beyond the limit.

Reply:
This is just to be stored as a result of the trying to call the WH endpoint; `requestUrlParsed` is not used to make the actual request.

We built `limitStringLength()` because we had situations where the HTTP error was larger than 255 chars, when trying to store the status on the `settings` table.

Currently, the webhook URL is a regular `varchar(255)` column, so it can't store more than that. I added an item to the [WH improvements ticket](https://github.com/metriport/metriport-internal/issues/1030) to work on that.

Reply:
Ok works for me